src/components/CourseItemDialog.tsx

```diff
@@ -175,6 +175,17 @@ const CourseItemDialog: React.FC<CourseItemDialogProps> = ({
                 activeItem.questions = [];
             }
         } else if (isOpen) {
+            // Add a new history entry when dialog opens to intercept back button
+            const hasChanges = activeItem.type === 'material'
+                ? learningMaterialEditorRef.current?.hasChanges() || false
+                : quizEditorRef.current?.hasChanges() || false;
+            
+            setTimeout(() => {
+                if (isEditMode || activeItem?.status === 'draft' || hasChanges) {
```

Comment:
why is `||` being used? If there are no changes and user is in edit mode, why should we prevent?